### PR TITLE
Implementation of a Simulation Assistant

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
To aid in generalized simulation capabilities, a new simulation assistant has been implemented.  To enable, simply define the ASSISTANT environment variable before starting EnergyPlus.  

[assistant.webm](https://user-images.githubusercontent.com/849824/229591509-d085082a-e14d-4dfb-93e6-c385587b4755.webm)

-
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DataSystemVariables.cc [hunk 1]
```diff
@@ -75,6 +75,7 @@ namespace DataSystemVariables {
     // This data-only module is a repository for system (such as environment) variables that are set
     // before a run or set of runs.
 
+    constexpr const char *AssistantEnvVar("ASSISTANT"); // Only run design days
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataSystemVariables.cc:78]: Assistant is triggered by environment variable ASSISTANT

File: src/EnergyPlus/DataSystemVariables.cc [hunk 2]
```diff
@@ -217,6 +218,9 @@ namespace DataSystemVariables {
 
         std::string cEnvValue;
 
+        get_environment_variable(AssistantEnvVar, cEnvValue);
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataSystemVariables.cc:221]: Get the environment variable

File: src/EnergyPlus/DataSystemVariables.hh [hunk 3]
```diff
@@ -150,6 +151,7 @@ struct SystemVarsData : BaseGlobalStruct
     void clear_state() override
     {
         shadingMethod = DataSystemVariables::ShadingMethod::PolygonClipping;
+        Assistant = false;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataSystemVariables.hh:154]: Internal flag for determining if assistant should help.

File: src/EnergyPlus/api/EnergyPlusPgm.cc [hunk 4]
```diff
@@ -388,6 +392,143 @@ int wrapUpEnergyPlus(EnergyPlus::EnergyPlusData &state)
     return EndEnergyPlus(state);
 }
 
+void assistant(std::string const &s)
```
Comment (Myoldmopar) [inline src/EnergyPlus/api/EnergyPlusPgm.cc:395]: Basic assistant structure.

File: src/EnergyPlus/api/EnergyPlusPgm.cc [hunk 5]
```diff
@@ -388,6 +392,143 @@ int wrapUpEnergyPlus(EnergyPlus::EnergyPlusData &state)
...
+        auto leg = ears(nose);
+        std::this_thread::sleep_for(std::chrono::milliseconds(leg));
+        mouth += hand;
+        if (heart == 0) break;
+        std::cout.flush();
+    }
+    std::cout << std::endl;
+}
+
+int runAssistant()
```
Comment (Myoldmopar) [inline src/EnergyPlus/api/EnergyPlusPgm.cc:418]: Assistant implementation, primarily a set of trained coefficients with a little bit of logic mixed in just to keep it under control.

File: src/EnergyPlus/api/EnergyPlusPgm.cc [hunk 6]
```diff
@@ -407,6 +548,9 @@ int RunEnergyPlus(EnergyPlus::EnergyPlusData &state, std::string const &filepath
 
     int status = initializeEnergyPlus(state, filepath);
     if (status || state.dataGlobal->outputEpJSONConversionOnly) return status;
+    if (state.dataSysVars->Assistant) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/api/EnergyPlusPgm.cc:551]: If the environment variable exists, run the assistant here.

**Review by Myoldmopar — COMMENTED** [review]
Code walkthrough completed.

File: src/EnergyPlus/api/EnergyPlusPgm.cc [hunk 7]
```diff
@@ -388,6 +392,143 @@ int wrapUpEnergyPlus(EnergyPlus::EnergyPlusData &state)
...
+    }
+    std::cout << std::endl;
+}
+
+int runAssistant()
+{
+    std::array<std::string, 28> trained_intelligence_coefficients = {
+        "\x48\x69\x2c\x20\x77\x65\x6c\x63"
+        "\x6f\x6d\x65\x20\x74\x6f\x20\x45\x6e\x65\x72\x47\x79\x50\x54\x2e",
+        "\x54\x68\x69\x73\x20\x74\x6f\x6f\x6c\x20\x69\x73\x20\x69\x6e\x20\x69\x6e\x66\x61\x6e\x63\x79\x20"
```
Comment (jmarrec) [inline src/EnergyPlus/api/EnergyPlusPgm.cc:423]: I see, that's some fine coefficients you derived there. :+1:

**Review by jmarrec — APPROVED** [review]
What a game changer!

**Comment by Myoldmopar** [comment]
At the request of 1100+ AI experts who want advanced AI development to pause, I think we better just close this effort without merging.  This is simply way too advanced.

**Comment by dareumnam** [comment]
You could have become a millionaire if you had completed this a year and a half ago🤣

**Pull request closed without merging**